junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -88,28 +90,47 @@ private ExtensionRegistry(ExtensionRegistry parent) {
 	}
 
 	/**
-	 * Get all {@code Extensions} in this registry or one of its ancestors of
+	 * Stream all {@code Extensions} in this registry or one of its ancestors of
 	 * the specified type.
 	 *
 	 * @param extensionType the type of {@link Extension} to get
 	 * @see #getReversedExtensions(Class)
-	 * @see #stream(Class)
+	 * @see #getExtensions(Class)
 	 */
-	public <E extends Extension> List<E> getExtensions(Class<E> extensionType) {
-		List<E> extensions = new ArrayList<>();
-
-		if (this.parent != null) {
-			extensions.addAll(this.parent.getExtensions(extensionType));
+	public <E extends Extension> Stream<E> streamExtensions(Class<E> extensionType) {
```

Comment:
But then again, it seems to fit JUnit 5's code style more than `extensionsStream`, so I suppose I wouldn't mind either way.